packages/snaps-ui/src/nodes.ts

```diff
@@ -28,73 +64,87 @@ export enum NodeType {
   Text = 'text',
 }
 
-export const CopyableStruct = object({
-  type: literal(NodeType.Copyable),
-  text: string(),
-});
+export const CopyableStruct = assign(
+  LiteralStruct,
+  object({
+    type: literal(NodeType.Copyable),
+    value: string(),
+  }),
+);
 
 /**
  * Text that can be copied to the clipboard.
+ *
+ * @property type - The type of the node, must be the string 'copyable'.
+ * @property value - The text to be copied.
  */
 export type Copyable = Infer<typeof CopyableStruct>;
 
-export const DividerStruct = object({
-  type: literal(NodeType.Divider),
-});
+export const DividerStruct = assign(
+  NodeStruct,
+  object({
+    type: literal(NodeType.Divider),
+  }),
+);
 
 /**
  * A divider node, that renders a line between other nodes.
  */
 export type Divider = Infer<typeof DividerStruct>;
 
-export const HeadingStruct = object({
-  type: literal(NodeType.Heading),
-  text: string(),
-});
+export const HeadingStruct = assign(
+  LiteralStruct,
+  object({
+    type: literal(NodeType.Heading),
+    value: string(),
+  }),
+);
 
 /**
  * A heading node, that renders the text as a heading. The level of the heading
  * is determined by the depth of the heading in the document.
  *
  * @property type - The type of the node, must be the string 'text'.
- * @property text - The text content of the node, either as plain text, or as a
+ * @property value - The text content of the node, either as plain text, or as a
  * markdown string.
  */
 export type Heading = Infer<typeof HeadingStruct>;
 
 export const PanelStruct: Struct<Panel> = assign(
-  NodeStruct,
+  ParentStruct,
   object({
     type: literal(NodeType.Panel),
```

Comment:
You might want to restrict which kind of children are allowed in Panel.

For example, Panel probably shouldn't be able to contain other Panels.

Reply:
Why not?

Reply:
At some point we could allow panels to be either rows or columns for more customizability

Reply:
Is Panel a UI component with it's own border and design, or is it flow component that aligns it's children somehow without no UI design?

Because if it has a border, and you nest like 10 of those, it's going to look ugly.

Reply:
It currently has no design and is just a box.